src/kv/store.h

```diff
@@ -219,35 +219,76 @@ namespace kv
       return *result;
     }
 
-    std::unique_ptr<Snapshot> snapshot(Version v) override
+    void deserialize(std::unique_ptr<Snapshot>& snapshot)
     {
-      std::lock_guard<SpinLock> mguard(maps_lock);
-
-      if (v > current_version())
       {
-        throw ccf::ccf_logic_error(fmt::format(
-          "Attempting to snapshot at invalid version v:{}, current_version:{}",
-          v,
-          current_version()));
+        std::lock_guard<SpinLock> vguard(version_lock);
```

Comment:
It looks like these changes should be applied _after_ the snapshot has been applied.

Reply:
I am simplifying this but the order should not matter. We are skipping forward an arbitrary number of version so there should not be any other transactions in progress.